python/cugraph-dgl/cugraph_dgl/nn/conv/gatconv.py

```diff
@@ -149,9 +150,19 @@ def forward(
         if g.is_block:
             if max_in_degree is None:
                 max_in_degree = g.in_degrees().max().item()
-            _graph = ops.make_mfg_csr(
-                g.dstnodes(), offsets, indices, max_in_degree, g.num_src_nodes()
-            )
+
+            if max_in_degree < self.MAX_IN_DEGREE_MFG:
+                _graph = ops.make_mfg_csr(
+                    g.dstnodes(), offsets, indices, max_in_degree, g.num_src_nodes()
+                )
+            else:
+                offsets_fg = torch.empty(
```

Comment:
where is there a difference between the offsets?

Reply:
offsets_mfg has a shape of (n_out_nodes+1, ), while offsets_fg has a shape of (n_in_nodes+1).
Hence, we have to pad (n_in_nodes - n_out_nodes) elements at the end of offset to form offset_fg.

Reply:
ah yes, sure ..., the kernels technically would work with offsets to be `n_out_nodes + 1`, however, as we use the offsets currently to determine `n_nodes` in our bindings we need it for that, my bad

Reply:
Either way: for the current GraphCast involvement, we would need proper support for any kind of bipartite-graphs (full graphs are just a special case of these when it comes to message passing) so we might address this rather quickly

Reply:
> ah yes, sure ..., the kernels technically would work with offsets to be `n_out_nodes + 1`, however, as we use the offsets currently to determine `n_nodes` in our bindings we need it for that, my bad

Oh, I though this restriction comes all the way from the C++ APIs, as noted here:
https://github.com/rapidsai/cugraph-ops/blob/ddd07dded5ddfdc88f22121b97adb9cc3ca4dbbd/cpp/include/cugraph-ops/graph/format.hpp#L82-L87

Regardless, support for generic bipartite graphs would be extremely beneficial to upstream integration.

Reply:
It is formulated like that in the comment. However, in the kernels themselves, you often just iterate over the offests (i.e. the out-nodes). Buffers from in-nodes are usually indexed like `input[indices[neigh]]` where `neigh` is between `offsets[current_out_node]` and `offsets[current_out_node + 1]`
In our CSR representation, offsets inherently refer to the out-nodes and indices to the in-nodes.